.github/workflows/ci-qemu.yml

```diff
@@ -0,0 +1,41 @@
+name: Build PR
+
+on:
+  pull_request:
+    branches: [ main ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+
+      # Cache .kernel
+      - uses: actions/cache@v2
+        env:
+          cache-name: build-cache-kernel
+        with:
+          path: ~/kernel/
+          key: ${{ runner.os }}-pr-${{ env.cache-name }}-5.9.16
+          restore-keys: |
+            ${{ runner.os }}-pr-${{ env.cache-name }}-5.9.16
+
+      - name: Check kernel already exists
+        id: check_kernel_image_exists
+        uses: andstor/file-existence-action@v1
+        with:
+          files: "/home/runner/kernel/bzImage-5.9.16"
+
+      - name: Build kernel
+        if: steps.check_kernel_image_exists.outputs.files_exists == 'false'
+        run: |
+          mkdir ~/kernel/
+          wget -q https://cdn.kernel.org/pub/linux/kernel/v5.x/linux-5.9.16.tar.xz
+          tar xf linux-5.9.16.tar.xz
+          cd linux-5.9.16 && cp /boot/config-`uname -r`* .config
```

Comment:
the problem is some new configs options are modules which means you have to add new modules  somehow which is hard..... so created a own kernel config which is much smaller to reduce the build time and all config option build into the image,
I'm using https://gist.githubusercontent.com/1Jo1/049b60e3ca767e9ad71c676f81f47add/raw/a71dce14d2e16e1944fefd66fa108193a53c466c/linux_5_9-ubu.config as a base config, this should work on ubuntu 20.04, tested on different kernels 5.10/5.11 etc.

Reply:
only works on x86, I'm gonna create a  config for ARM ubuntu as well, probably tomorrow